src/components/Icon/index.js

```diff
@@ -1,29 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 
 function Icon({
   className,
   family,
   type,
+  hoverIconType,
   ...rest
 }) {
-  function getCSSClasses() {
+  const [isHovered, setIsHovered] = useState(false);
+  function getCSSClasses(showHoverIcon) {
     const fontFamily = family || 'fa';
 
     return [
       'tyk-icon',
       fontFamily,
-      `${fontFamily}-${type}`,
+      `${fontFamily}-${showHoverIcon ? hoverIconType : type}`,
```

Comment:
I think that the decision of what classes to add for the component should be taken in this function instead of the markup.
```suggestion
      `${fontFamily}-${isHovered && hoverIconType ? hoverIconType : type}`,
```